spring-cloud-starter-stream-sink-log/src/main/java/org/springframework/cloud/stream/app/log/sink/LogSinkConfiguration.java

```diff
@@ -40,7 +47,20 @@
 	@Bean
 	@ServiceActivator(inputChannel = Sink.INPUT)
 	public LoggingHandler logSinkHandler() {
-		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name());
+		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name()) {
+
+			@Override
+			protected void handleMessageInternal(Message<?> message) throws Exception {
```

Comment:
I would prefer to move all the logic to the `setLogExpression()`.
There is a `FunctionExpression` for the consideration.

Also it looks like we could use `ObjectToStringTransformer` in front of this `LoggingHandler` if that is a case.
The `contentType` could be consulted from the `Filter` as well.

Just another variant to achieve a requirements.